### PR TITLE
kvclient: add timeseries for all error details

### PR DESCRIPTION
**Pull request by andreimatei**
This patch adds timeseries the count of errors of each type received by
the DistSender. It should be useful to figure out what error a node is
receiving; there previously was very poor visibility into that.

Release note: None

### CODE REVIEW

**Review by ajwerner — COMMENTED** [review]
Generally strong support from me!

> Reviewed 1 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-:-METWSOP95MtQPny8VNW:bjs5u9p)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @andreimatei, @nvanbenschoten, and @tbg)

___
*[pkg/kv/kvclient/kvcoord/dist_sender.go, line 1896 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-METWP8h7_32ixTKD8vw:-METWP8h7_32ixTKD8vx:b2s2ria) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/kv/kvclient/kvcoord/dist_sender.go#L1896)):*
> ```Go
> 			}
> 		} else {
> 			br.Error.GoError()
> ```

What is this?
___
*[pkg/roachpb/errors.go, line 233 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-METWFao3x70zJXhr_vJ:-METWFapEQWc6NxnImVS:b-5gllag) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/roachpb/errors.go#L233)):*
> ```Go
> 	InternalErrType ErrorDetailType = 25
> 
> 	NumErrors ErrorDetailType = 40
> ```

nit: make this an `int`

File: pkg/kv/kvclient/kvcoord/dist_sender.go [hunk 1]
```diff
@@ -189,6 +196,13 @@ func makeDistSenderMetrics() DistSenderMetrics {
 		meta.Help = fmt.Sprintf(meta.Help, method)
 		m.MethodCounts[i] = metric.NewCounter(meta)
 	}
+	for i := range m.ErrCounts {
+		errType := roachpb.ErrorDetailType(i).String()
+		meta := metaDistSenderErrCountTmpl
+		meta.Name = fmt.Sprintf(meta.Name, strings.ToLower(errType))
+		meta.Help = fmt.Sprintf(meta.Help, errType)
+		m.MethodCounts[i] = metric.NewCounter(meta)
```
Comment (tbg) [inline pkg/kv/kvclient/kvcoord/dist_sender.go:204]: copy pasta error

File: pkg/kv/kvclient/kvcoord/dist_sender.go [hunk 2]
```diff
@@ -1878,6 +1893,7 @@ func (ds *DistSender) sendToReplicas(
 				}
 			}
 		} else {
+			br.Error.GoError()
```
Comment (tbg) [inline pkg/kv/kvclient/kvcoord/dist_sender.go:1896]: ?

File: pkg/roachpb/errors.go [hunk 3]
```diff
@@ -904,6 +1061,10 @@ func (e *RangeFeedRetryError) message(pErr *Error) string {
 	return fmt.Sprintf("retry rangefeed (%s)", e.Reason)
 }
 
+func (e *RangeFeedRetryError) Type() ErrorDetailType {
```
Comment (tbg) [inline pkg/roachpb/errors.go:1064]: Must've been fun to type these out!

**Comment by tbg** [comment]
👍

**Review by tbg — COMMENTED** [review]
Looks good, but also a bit of cleanup left, and a test would be appreciated.

**Review by andreimatei — COMMENTED** [review]
added a test

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-:-MF27Jk7E-YkvG8ez4At:bhnwd5i)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @nvanbenschoten, and @tbg)

___
*[pkg/kv/kvclient/kvcoord/dist_sender.go, line 204 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-ME43kzbVA34PpT7kvgU-r1-204:-MF1mSAB-2A5JKXud57u:b3175tm) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/kv/kvclient/kvcoord/dist_sender.go#L204)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

copy pasta error
</blockquote></details>

oops done
___
*[pkg/kv/kvclient/kvcoord/dist_sender.go, line 1896 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-METWP8h7_32ixTKD8vw:-MF1mWY-0PL71ItiMty1:bb7gzl0) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/kv/kvclient/kvcoord/dist_sender.go#L1896)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

What is this?
</blockquote></details>

gone
___
*[pkg/kv/kvclient/kvcoord/dist_sender.go, line 1896 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-ME43kzbVA34PpT7kvgU-r1-1896:-MF1mXGFEHY5D_R11cfe:bb7gzl0) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/kv/kvclient/kvcoord/dist_sender.go#L1896)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

?
</blockquote></details>

gone
___
*[pkg/roachpb/errors.go, line 233 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-METWFao3x70zJXhr_vJ:-MF1nBhK3lSjbAWj5sHz:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/roachpb/errors.go#L233)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

nit: make this an `int` 
</blockquote></details>

done
___
*[pkg/roachpb/errors.go, line 1064 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-ME43kzbVA34PpT7kvgV-r1-1064:-MF1nDOOCPtO27qjrVe-:b-k1dbez) ([raw file](https://github.com/cockroachdb/cockroach/blob/d66a9e795fd2393382cca85770a845560059123f/pkg/roachpb/errors.go#L1064)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Must've been fun to type these out!
</blockquote></details>

super fun

**Review by ajwerner — COMMENTED** [review]
:lgtm: This is very likely to become my new favorite custom chart.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-:-MF2ZC1C01ssNy0GF_nd:b-n1jwvu)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @nvanbenschoten, and @tbg)

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52490#-:-MF7grugAwSSglGQ03mn:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner, @nvanbenschoten, and @tbg)

**Pull request merged**